go-controller/pkg/util/multi_network.go

```diff
@@ -662,19 +662,56 @@ func GetPodNADToNetworkMapping(pod *kapi.Pod, nInfo NetInfo) (bool, map[string]*
 	for _, network := range allNetworks {
 		nadName := GetNADName(network.Namespace, network.Name)
 		if nInfo.HasNAD(nadName) {
+			if nInfo.IsPrimaryNetwork() {
+				return false, nil, fmt.Errorf("unexpected primary network %q specified with a NetworkSelectionElement %+v", nInfo.GetNetworkName(), network)
+			}
 			if _, ok := networkSelections[nadName]; ok {
 				return false, nil, fmt.Errorf("unexpected error: more than one of the same NAD %s specified for pod %s",
 					nadName, podDesc)
 			}
 			networkSelections[nadName] = network
 		}
 	}
+
 	if len(networkSelections) == 0 {
 		return false, nil, nil
 	}
+
 	return true, networkSelections, nil
 }
 
+func GetPodNADToNetworkMappingWithActiveNetwork(pod *kapi.Pod, nInfo NetInfo, activeNetwork NetInfo) (bool, map[string]*nettypes.NetworkSelectionElement, error) {
+	on, networkSelections, err := GetPodNADToNetworkMapping(pod, nInfo)
+	if err != nil {
+		return false, nil, err
+	}
+
+	if activeNetwork == nil {
+		return on, networkSelections, nil
+	}
+
+	if activeNetwork.IsDefault() ||
+		activeNetwork.GetNetworkName() != nInfo.GetNetworkName() ||
+		nInfo.TopologyType() == types.LocalnetTopology {
+		return on, networkSelections, nil
+	}
+
+	// Add the active network to the NSE map if it is configured
+	activeNetworkNADs := activeNetwork.GetNADs()
+	if len(activeNetworkNADs) < 1 {
+		return false, nil, fmt.Errorf("missing NADs at active network '%s' for namesapce '%s'", activeNetwork.GetNetworkName(), pod.Namespace)
+	}
+	activeNetworkNADKey := strings.Split(activeNetworkNADs[0], "/")
```

Comment:
Can you actually assume the **correct** NAD name to be the first NAD in the NADs list  ?

Let's say I have a network spanning 2 namespaces - `tenantblue` and `tenantyellow` ; the tenants then spin up 2 pods (1 pod in each namespace).

The active network NAD name for the attachment on the tenantblue pod would be `tenantblue/<metadata nad name>` while for `tenantyellow` it would have to be `tenantyellow/<metadata nad name>`. 

With your code, both would have the same NAD name. That would error on all pods not belonging to that namespace, wouldn't it ?

Reply:
Can you add a unit test with a network spanning 2 namespaces ? (somewhere).

Reply:
OK, I have better understood how this goes - we are **not** returning the network controller net info, but creating a **new** `NetInfo` from a single NAD (the correct one).

Thus, we're OK.